speaker/bgp_controller.go

```diff
@@ -335,13 +335,27 @@ func (c *bgpController) updateAds() error {
 		if peer.session == nil {
 			continue
 		}
-		if err := peer.session.Set(allAds...); err != nil {
+		ads := adsPerPeer(peer.cfg.Name, allAds)
+		if err := peer.session.Set(ads...); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func adsPerPeer(peerName string, ads []*bgp.Advertisement) []*bgp.Advertisement {
```

Comment:
nit: rename to `adsForPeer` (this reads a bit like it'll return a map or something)